src/components/select/SelectCommons.tsx

```diff
@@ -0,0 +1,329 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+import styled from '@emotion/styled';
+import { Select as MantineSelect } from '@mantine/core';
+import { ComponentProps, forwardRef } from 'react';
+import { useIntl } from 'react-intl';
+import { isDefined } from '~common/helpers/types';
+import { PropsWithLabels } from '~types/utils';
+import { IconChevronDown, Spinner } from '..';
+import { INPUT_SIZE_VALUES, InputSize } from '../../utils/inputs';
+import { useSelectItemComponent } from './SelectItemCommons';
+import { SelectHighlight, SelectOption, SelectOptionType } from './SelectTypes';
+
+type MantineSelectProps = ComponentProps<typeof MantineSelect>;
+export interface SelectBaseProps {
+  className?: string;
+  data: ReadonlyArray<SelectOption>;
+  defaultValue?: MantineSelectProps['defaultValue'];
+  filter?: MantineSelectProps['filter'];
+  hasError?: boolean;
+  highlight?: SelectHighlight;
+  isDisabled?: boolean;
+  isLoading?: boolean;
+  isNotClearable?: boolean;
+  isSearchable?: boolean;
+  isRequired?: boolean;
+  labelError?: MantineSelectProps['error'];
+  labelNotFound?: MantineSelectProps['nothingFound'];
+  limit?: MantineSelectProps['limit']; // might change for a max height
+  name?: MantineSelectProps['name'];
+  optionComponent?: MantineSelectProps['itemComponent'];
+  optionType?: SelectOptionType;
+  onChange: MantineSelectProps['onChange'];
+  onOpen?: MantineSelectProps['onDropdownOpen'];
+  onSearch?: MantineSelectProps['onSearchChange'];
+  placeholder?: MantineSelectProps['placeholder'];
+  size?: InputSize;
+  value: MantineSelectProps['value'];
+  valueIcon?: MantineSelectProps['icon'];
+}
+
+export const SelectBase = forwardRef<HTMLInputElement, PropsWithLabels<SelectBaseProps>>(
+  (props, ref) => {
+    const {
+      ariaLabel,
+      ariaLabelledBy,
+      data,
+      hasError = false,
+      helpText,
+      highlight = SelectHighlight.Default,
+      isDisabled = false,
+      isLoading = false,
+      isNotClearable = false,
+      isSearchable = false,
+      isRequired = false,
+      label,
+      labelError,
+      labelNotFound,
+      onSearch,
+      onOpen,
+      optionComponent,
+      optionType = SelectOptionType.Check,
+      size = InputSize.Full,
+      valueIcon,
+      ...selectProps
+    } = props;
+
+    const intl = useIntl();
+
+    const itemComponent = useSelectItemComponent(optionComponent, optionType);
+    const isClearable = !isNotClearable && !isRequired;
+
+    // TODO Highlighter for search
```

Comment:
if it's not done now it should probably be a ticket instead of a TODO comment

Reply:
https://sonarsource.atlassian.net/browse/DS-441 :+1: